docs/lang/articles/performance/performance.md

```diff
@@ -223,3 +223,26 @@ we found BLS to be more effective for caching the destinations of the atomic ope
 
 As a rule of thumb, run benchmarks to decide whether to enable BLS or not.
 :::
+
+## Offline Cache
+
+A taichi kernel is implicitly compiled the first time it is invoked. The compilation yieldings are kept in an "online" in-memory cache to reduce the overhead in subsequent function calls. As long as the kernel function is unchanged, it can be directly loaded and launched. However, the cache is no longer availabe when the entire program runs down to its termination. Taichi re-compiles all the kernel functions and constructs the "online" in-memory cache when the user runs the program again. The first launch of a Taichi function is always slow due to compilation overhead.
+
+We address the problem by introducing the "offline" cache feature, that could dump and save the compilation cache on disk for future runs. The first launch overhead can be drastically reduced in repeated runs. Taichi now automatically constructs and maintains the offline cache by default. We also provide several options in `ti.init` to configure the detailed behaviors of offline cache.
```

Comment:
```suggestion
We address this problem by introducing the *offline* cache feature, which dumps and saves the compilation cache on disk for future runs. The first launch overhead can be drastically reduced in repeated runs. Taichi now constructs and maintains an offline cache by default, as well as providing several options in `ti.init()` for configuring the offline cache behavior.
```

Reply:
I'm not quite certain about "The first launch overhead can be drastically reduced in repeated runs.". Could I ask you to elaborate on this one? I thought the first launch overhead would stay unchanged and would not be reduced?

Reply:
The overhead of "the first launch" includes compilation and the truly 1st run. If the launched Taichi kernel's compilation data was saved on disk, the next "the first launch" overhead would be reduced because of **offline cache**.